src/lib/consensus/intf.ml

```diff
@@ -145,4 +145,13 @@ module type S = sig
   (**
    * A hook for managing local state when the locked tip is updated.
    *)
+
+  val should_bootstrap :
+       max_length:int
```

Comment:
This is an inversion of concerns. The `max_length` here represents the `blocks_till_finality`, which is a property provided by the consensus mechanism, not to it. Instead, you should expose a `blocks_till_finality` value from the consensus interface and use that to determine the size of the transition frontier, not the other way around.